src/plugins/intel_cpu/src/nodes/inverse.cpp

```diff
@@ -0,0 +1,257 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "inverse.hpp"
+
+#include <openvino/core/type.hpp>
+#include <openvino/op/constant.hpp>
+#include <openvino/op/inverse.hpp>
+
+#include "utils/bfloat16.hpp"
+
+// Parallel LU decomposition algorithm with partial pivoting
+// Based on the lectures by Prof. Dr. Thomas Huckle, Parallel Numerics
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+Inverse::Inverse(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr& context)
+    : Node(op, context, NgraphShapeInferFactory(op, EMPTY_PORT_MASK)) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        THROW_CPU_NODE_ERR(errorMessage);
+    }
+
+    auto inverse_op = as_type_ptr<op::v14::Inverse>(op);
+    m_adjoint = inverse_op->get_adjoint();
+
+    constant = ConstantType::StrictNoConst;
+
+    m_const_input = is_type<op::v0::Constant>(op->get_input_node_ptr(INPUT_PORT));
+}
+
+bool Inverse::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (op->get_type_info() != op::v14::Inverse::get_type_info_static()) {
+            errorMessage = "Only Inverse operation from the opset14 is supported by the CPU plugin.";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+void Inverse::getSupportedDescriptors() {
+    if (getParentEdges().size() != 1) {
+        THROW_CPU_NODE_ERR("has incorrect number of input edges.");
+    }
+    if (getChildEdges().empty()) {
+        THROW_CPU_NODE_ERR("has incorrect number of output edges.");
+    }
+}
+
+void Inverse::initSupportedPrimitiveDescriptors() {
+    m_input_precision = getOriginalInputPrecisionAtPort(INPUT_PORT);
+    if (!one_of(m_input_precision, ov::element::f32, ov::element::f16, ov::element::bf16)) {
+        m_input_precision = ov::element::f32;
+    }
+
+    addSupportedPrimDesc({{LayoutType::ncsp, m_input_precision, m_const_input}},
+                         {{LayoutType::ncsp, m_input_precision}},
+                         ref_any);
+}
+
+bool Inverse::needShapeInfer() const {
```

Comment:
Why do we need to execute shape inference every time? It looks like there is no need to override functions needShapeInfer(), needPrepareParams() and isExecutable().

Reply:
After checking the default implementations, I found ou that indeed they are useless. Removed!